widgetastic_manageiq.py

```diff
@@ -878,3 +879,280 @@ def last_page(self):
+    """
+    def __init__(self, parent, locator, logger=None):
+        Widget.__init__(self, parent=parent, logger=logger)
+        self._locator = locator
+
+    def __locator__(self):
+        return self._locator
+
+    @property
+    def button_names(self):
+        buttons = self.browser.elements('//label[input[@type="radio"]]')
+        return [self.browser.text(btn) for btn in buttons]
+
+    @property
+    def buttons(self):
+        return [RadioButton(self, name=name) for name in self.button_names]
+
+    @property
+    def selected(self):
+        return [btn for btn in self.buttons if btn.is_selected][-1].name
+
+    def select(self, name):
+        button = [btn for btn in self.buttons if btn.name == name][-1]
+        if not button.is_selected:
+            button.click()
+            return True
+        return False
+
+    def read(self):
+        return self.selected
+
+    def fill(self, name):
+        return self.select(name)
+
+
+class BreadCrumb(Widget):
+    """ CFME BreadCrumb navigation control
+
+        .. code-block:: python
+        breadcrumb = BreadCrumb()
+        breadcrumb.click_location(breadcrumb.locations[0])
+    """
+    ROOT = '//ol[@class="breadcrumb"]'
+    ELEMENTS = './/li'
+
+    def __init__(self, parent, locator=None, logger=None):
+        Widget.__init__(self, parent=parent, logger=logger)
+        self._locator = locator or self.ROOT
+
+    def __locator__(self):
+        return self._locator
+
+    @property
+    def _path_elements(self):
+        return self.browser.elements(self.ELEMENTS)
+
+    @property
+    def locations(self):
+        return [self.browser.text(loc) for loc in self._path_elements]
+
+    @property
+    def active_location(self):
+        br = self.browser
+        return [br.text(loc) for loc in self._path_elements if 'active' in br.classes(loc)][-1]
+
+    def click_location(self, name):
+        br = self.browser
+        location = [loc for loc in self._path_elements if br.text(loc) == name][-1]
+        return br.click(location)
+
+
+class ToolBarViewSelector(View):
+    """
+    represents toolbar's view selector control.
+    """
+    list_view = Button(title='List View')
+    grid_view = Button(title='Grid View')
+    tile_view = Button(title='Tile View')
+
+    def __locator__(self):
+        return './/div[contains(@class, "toolbar-pf-view-selector")]'
+
+    def select(self, item):
+        if item == self.list_view.title:
+            self.list_view.click()
+        elif item == 'Grid View':
+            self.grid_view.click()
+        elif item == 'Tile View':
+            self.grid_view.click()
+        else:
+            raise ValueError('Incorrect value passed')
+
+    @property
+    def selected(self):
+        """
+        goes thru buttons and returns the title of active button
+        Returns: currently selected view
+
+        """
+        return [btn.title for btn in (self.list_view, self.grid_view, self.tile_view)
+                if btn.active][-1]
```

Comment:
If you are sure that there will be only these three widgets placed on this view, you can pretty much do just `for btn in self` since iterating a View yields widgets in the definition order :)